generator/app.go

```diff
@@ -105,33 +101,81 @@ func createDirIfNotExists(path string) error {
 	return nil
 }
 
+func getExistingContent(path string) (string, error) {
+	capture := false
+	var captured []string
+
+	// NOTE: For some reason using bufio.Scanner with existing file pointer prepends
+	// a bunch of null ^@ characters, so using to ioutil.ReadFile instead.
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+
+	for _, line := range strings.Split(string(content), "\n") {
+		if strings.Contains(line, endMarker) {
+			capture = false
+		}
+		if capture {
+			captured = append(captured, line)
+		}
+		if strings.Contains(line, beginMarker) {
+			capture = true
+		}
+	}
+
+	return strings.Join(captured, "\n"), nil
+}
+
 func writeTemplate(templatePath, outputPath string, data interface{}) error {
 	// set up template
-	t, err := template.ParseFiles(templatePath, headerTemplate, footerTemplate)
+	t, err := template.ParseFiles(templatePath, headerTemplate)
 	if err != nil {
 		return err
 	}
 
 	// open file and truncate
-	f, err := os.OpenFile(outputPath, os.O_WRONLY, 0644)
+	f, err := os.OpenFile(outputPath, os.O_RDWR, 0644)
 	if err != nil {
 		return err
 	}
 	defer f.Close()
+
+	// get any existing content
+	content, err := getExistingContent(outputPath)
+	if err != nil {
+		return err
+	}
+
+	// ensure file is empty
 	f.Truncate(0)
 
-	// write template output to file
+	// generated content
```

Comment:
Nit: I would have passed existing content into the template vs. having three functions append to the same file.  But this works well too.